### PR TITLE
Add validate-db-reset.jenkinsfile

### PR DESCRIPTION
**Pull request by cabrador**
This PR adds `validate-db-reset.jenkinsfile` which tests the `reset` feature in

### CODE REVIEW

**Review by kjezek — CHANGES_REQUESTED** [review]
Please use a "standard" way of checkout and build of Aida+Carmen, it makes sure a compatible composite is tested. 

```

        stage('Checkout') {
            steps {
                script {
                    currentBuild.description = "Building on ${env.NODE_NAME}"
                }

                checkout scmGit(
                    branches: [[name: "${AidaVersion}"]],
                    userRemoteConfigs: [[
                        url: 'https://github.com/Fantom-foundation/Aida.git'
                    ]]
                )
                sh "git submodule update --init --recursive"

                dir('carmen') {
                    checkout scmGit(
                        branches: [[name: "${CarmenVersion}"]],
                        userRemoteConfigs: [[url: 'https://github.com/Fantom-foundation/Carmen.git']]
                    )
                }
            }
        }

        stage('Build') {
            steps {
                sh "go mod tidy"
                sh "make -j aida-vm-sdb"
                sh 'go test ./...'
            }
        }

```

File: carmen/validate-db-reset.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        timeout (time: 10, unit: 'HOURS') // ~7 hours
+        disableConcurrentBuilds(abortPrevious: false)
+    }
+
+    environment {
+        GOROOT = '/usr/lib/go-1.21/'
+        GOGC = '50'
+        GOMEMLIMIT = '120GiB'
+        AIDADB = '/mnt/aida-db-central/aida-db'
+	    TMPDB = '/mnt/tmp-disk'
```
Comment (Lubomir-Jahn) [inline carmen/validate-db-reset.jenkinsfile:15]: nit: formatting
```suggestion
        TMPDB = '/mnt/tmp-disk'
```

**Review by Lubomir-Jahn — APPROVED** [review]
Just please fix the formatting. Looks good otherwise 👍🏻

**Review by kjezek — CHANGES_REQUESTED** [review]
It looks good to me, just the pipeline seems to manipulate files that are never created by this pipeline - I guess copy+paste error. It should be cleared up, not to confuse the reader.

File: carmen/validate-db-reset.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            steps {
+                sh "go mod tidy"
+                sh "make -j aida-vm-sdb"
+                sh 'go test ./...'
+            }
+        }
+
+        stage('Loop') {
+            steps {
+                sh "rm -f *.cpuprofile *.memprofile *.log"
```
[the remark this answers is not in the document]
Comment (Lubomir-Jahn) [reply carmen/validate-db-reset.jenkinsfile:58]: Good point. These files are really not produced by this pipeline and this line should be removed.
```suggestion
```

**Review by kjezek — CHANGES_REQUESTED** [review]
General feedback first:
after previous round of review, the pipeline was basically ready. For a next review round, the way the pipeline works is widely changed. This makes review process and generally path to the final solution very difficult. The review should lead step-by-step to the final version, but when a step is complete rework, it can hardly lead to a quick resolution of the task.

Now to the pipeline: it executes 3 times exactly the same code. I am not fan of this approach. What if we want to scale this pipeline from 0 to N, do we want to end-up changing the pipeline every time?

What if we have just the simple pipeline with one job as it was previously and only plan as many jobs as we want in parallel? Lets say this job takes one hour, we can then schedule this job to start every 20minutes, and we would reach the same scaling. Of course, we can modify this as frequently as we like.

**Review by kjezek — APPROVED** [review]
much more elegant, thanks

**Pull request merged**